src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+type Props = {
+  onAddTodo: (title: string) => void;
+  isDisabled: boolean;
+  allTodoCompleted: boolean;
+  hasTodo: boolean;
+};
+
+export const TodoForm: React.FC<Props> = React.memo(({
+  onAddTodo,
+  isDisabled,
+  allTodoCompleted,
+  hasTodo,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const titleInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (hasTodo) {
+      titleInput.current?.focus();
+    }
+  }, [isDisabled]);
```

Comment:
Don't you need to include `hasTodo` in dependencies?

Reply:
If i include hasTodo in dependencies the input field will be in focus immediately

Reply:
In this case, there should be one more variable to decide if you want to focus it
because current behavior is a bit unexpected when you read code

It is not a critical thing I guess - try to fix it in the next part of the task